File: SupermarketAPI/Models/Supplier.cs
using System.ComponentModel.DataAnnotations;

namespace SupermarketAPI.Models
{
    public class Supplier
    {
        public int Id { get; set; }

        [Required]
        [StringLength(255)]
        public string Name { get; set; } = string.Empty;

        [StringLength(20)]
        public string? ContactNo { get; set; }

        [StringLength(255)]
        public string? Address { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public bool IsActive { get; set; } = true;
    }
}

File: supermarket_flutter_app/lib/features/supplier/models/supplier.dart
class Supplier {
  final int id;
  final String name;
  final String? contactNo;
  final String? address;

  Supplier({
    required this.id, 
    required this.name, 
    this.contactNo, 
    this.address
  });

  factory Supplier.fromJson(Map<String, dynamic> json) {
    return Supplier(
      id: json['id'],
      name: json['name'],
      contactNo: json['contactNo'],
      address: json['address'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'contactNo': contactNo,
      'address': address,
    };
  }
}
